api/src/services/users/create_api_key.py

```diff
@@ -46,6 +53,24 @@ def create_api_key(db_session: db.Session, user_id: UUID, json_data: dict) -> Us
     )
     db_session.add(api_key)
 
+    # Import the API key to AWS API Gateway
+    try:
+        _import_api_key_to_aws_gateway(api_key)
+    except Exception as e:
+        # Remove the API key from the database since AWS API Gateway integration failed
+        # This ensures consistency - if the key can't be created in API Gateway,
+        # it shouldn't exist in our database either
+        db_session.rollback()
```

Comment:
If this logic is wrapped in a `with db_session.begin()` then we don't need to do a rollback as that already makes a transaction that wasn't committed. When the `with` block ends it'll commit. If there is any error, it already does the rollback.